src/db/serde.ts

```diff
@@ -0,0 +1,98 @@
+import { Column, Insertable, Table } from "zapatos/schema";
+
+export interface Hook<U, T> {
+  [t: Table]: { [c: Column]: (x: U) => T };
+}
+
+// TODO: narrow these types
+const DESERIALIZE_HOOK: Hook<any, any> = {};
+const SERIALIZE_HOOK: Hook<any, any> = {};
+
+function applyHook<U, T>(
+  hook: Hook<U, T>,
+  table: Table,
+  values: Insertable | Insertable[]
+): Insertable | Insertable[] {
+  return Array.isArray(values)
+    ? values.map((v) => applyHookSingle(hook, table, v))
+    : applyHookSingle(hook, table, values);
+}
+
+function applyHookSingle<U, T>(
+  hook: Hook<U, T>,
+  table: Table,
+  values: Insertable
+): Insertable {
+  const processed: Insertable = {};
+  for (const [k, v] of Object.entries(values)) {
+    const f = hook?.[table]?.[k];
+    processed[k] = f ? f(v) : v;
+  }
+  return processed;
+}
+
+function registerHook<U, T>(
+  hook: Hook<U, T>,
+  table: Table,
+  column: Column,
+  f: (x: U) => T
+): void {
+  if (!(table in hook)) {
+    hook[table] = {};
+  }
+  hook[table][column] = f;
+}
+
+export function applyDeserializeHook(
+  table: Table,
+  values?: Insertable | Insertable[]
+): undefined | Insertable | Insertable[] {
+  if (!values) {
+    return values;
+  }
+  return applyHook(DESERIALIZE_HOOK, table, values);
+}
+
+export function applySerializeHook(
+  table: Table,
+  values: Insertable | Insertable[]
+): Insertable | Insertable[] {
+  return applyHook(SERIALIZE_HOOK, table, values);
+}
+
+// TODO: f should only read native types
+export function registerDeserializeHook<T>(
+  table: Table,
+  column: Column,
+  f: (x: any) => T
+) {
+  registerHook(DESERIALIZE_HOOK, table, column, f);
+}
+
+// TODO: f should only return native types
```

Comment:
what are the native types? native to ts? postgres? json? regardless, could start with a `NativeType` type and use that

Reply:
in Zapatos langauge, the `baseTsType`. There's not an inventory of these types rn, I think. There's a list of strings in `pgTypes.ts`. Before adding `NativeType`, I think we should take a longer look at what the types should actually be.